src/store/creator/selectors.ts

```diff
@@ -1,12 +1,12 @@
 import { AppState } from '../types';
-import { CreatorItem, CreatorStatus } from './types';
+import { CreatorItem, CreatorStagedItem, CreatorStatus } from './types';
 
 type State = Pick<AppState, 'creator'>;
 
-export const getCreatorStaged = (state: State): CreatorItem | null => state.creator.staged;
+export const getCreatorStaged = (state: State): CreatorStagedItem | null => state.creator.staged;
 export const getCreatorStagedForLocation = (state: State, location: number): CreatorItem | null => {
     const staged = getCreatorStaged(state);
-    return staged && staged.location === location ? staged : null;
+    return staged?.shape && staged.location === location ? (staged as CreatorItem) : null;
```

Comment:
The casting and shape check lead to confusing behavior for this selector. It should return the staged value if the location matches.